File: app/widgets/chromosome/probes/catalogs/action_panel.js
//= require <templates/action_panel>
//= require <chromosome/probe>

Module('Chromosome.Probes.Catalogs', function () {
  JooseClass('ActionPanel', {
    isa: Templates.ActionPanel,
    has: {
      buttons: {
        is: 'ro',
        init: function () {
          return [

          {
            label: 'Create',
            img: {
              src: '/images/small_addnew.gif'
            },
            imode: 'edit'
          },
          {
            label: 'Delete',
            img: {
              src: '/images/small_cross.png'
            },
            imode: 'edit'
          }
          ]
        }
      },
      catalog: {
        is: 'ro',
        init: function () {
          return this.parent()
        }
      }
    },
    methods: {

      onClick: function(event) {
        var me = this;
        Event.delegate({
          'input[type="button"]': function (event) {

          switch (event.element().readAttribute('value')) {
        

            case 'Create':
              this.viewport().widget('window').loadPage('new_project_chromosome_probe_path', {
                probe_id: this.params().id
              })
              break;
              
            case 'Delete':
              this.requireSelection(function(){
                var probe = new Chromosome.Probe({context: this.context()});
                probe.deleteSelected({collectionString: "probe(s)"});
              })
              break;
          }
        }
      }).call(this,event)
    }
    }
  })
});

File: app/widgets/chromosome/probes/catalogs/action_panel.rb
class Chromosome::Probes::Catalogs::ActionPanel < Templates::ActionPanel
  def initialize options
    options = { buttons: [ { label: 'Create', img: { src: '/images/small_addnew.gif' } },
                           { label: 'Delete', img: { src: '/images/small_cross.png' } },
                           { label: 'Bulk Upload', img: { src: "/images/small_addnew.gif" }, imode: 'edit' }

#                           { label: 'Align', img: { src: '/images/addnew.gif'} }
                         ]}.merge(options)
    super
    if interact_mode != 'edit'
      @buttons = {}
    end
  end
end
